Reorganize HTMLElement.dir and HTML dir attr references (#35683)
mdn/content: files/en-us/web/api/htmlelement/dir/index.md

{{ApiRef("HTML DOM")}}

> The **`HTMLElement.dir`** property indicates the text writing directionality of the content of the current element. It reflects the element's [`dir`](/en-US/docs/Web/HTML/Global_attributes/dir) attribute.

Reviewer: I would still state that the property is used for getting/setting the value, and note that if the attribute is not implicitly set then `HTMLElement.dir` will return an empty value even if it has inherited from it's parent.

Author: How do you "implicitly set" an attribute? The default value being "" is already stated in the "value" section. Also "it reflects the attribute" is exactly equivalent to "it gets and sets the attribute".

Reviewer: It appears that my dyslexia has struck again. I mean explicitly, for example:
`<p>My paragraph.</p>` this equates to `dir=""` and `HTMLElement.dir` returns `""`
`<p dir="auto">My paragraph.</p>` and `HTMLElement.dir` returns `"auto"`

Author: Doesn't that just mean it simply reflects the attribute value instead of representing the real directionality used during render, as already stated in this description?

Reviewer: Thank you

> The **`HTMLElement.dir`** property indicates the text writing directionality of the content of the current element. It reflects the element's [`dir`](/en-US/docs/Web/HTML/Global_attributes/dir) attribute. Note that even if the `dir` attribute is unspecified, the element may still inherit directionality from its parent, which is not reflected by this property's value.

Reviewer 2: Perhaps even more "highlighted".

```suggestion
The **`HTMLElement.dir`** property indicates the text writing directionality of the content of the current element.
It reflects the value, if any, explicitly set in the element's [`dir`](/en-US/docs/Web/HTML/Global_attributes/dir) attribute.

Note that if the `dir` attribute is unspecified, the element itself may still inherit directionality from its parent.
However that inherited directionality is not reflected by this property's value.
```

Author: It _always_ reflects the `dir` attribute. The `dir` attribute, if unspecified, defaults to `""`.

```suggestion
The **`HTMLElement.dir`** property indicates the text writing directionality of the content of the current element. It reflects the element's [`dir`](/en-US/docs/Web/HTML/Global_attributes/dir) attribute.

Note that if the `dir` attribute is unspecified, the element itself may still inherit directionality from its parent. However, that inherited directionality is not reflected by this property's value.
```

Reviewer 2: Thanks for the clarification. This is good.

The text writing directionality of an element is which direction that text goes (for support of different language systems). Arabic languages and Hebrew are typical languages using the RTL directionality.

## Value

One of the following:

- `"ltr"`, for left-to-right;
- `"rtl"`, for right-to-left;
- `"auto"` for specifying that the direction of the element must be determined based on the contents of the element.

> - `""`, the default value, which means the directionality is inherited from the parent element.

Reviewer 2: Can we make this a definition list and link to the items in various places?

Author: ```suggestion
- `"ltr"`
  - : Left-to-right writing direction.
- `"rtl"`
  - : Right-to-left writing direction.
- `"auto"`
  - : The direction of the element must be determined based on the contents of the element.
- `""`
  - : The default value; the directionality is inherited from the parent element.
```
